examples/nats-box/main.rs

```diff
@@ -28,12 +31,24 @@ enum Command {
 
 fn main() -> CliResult {
     let args = Cli::from_args();
-    let nc = nats::ConnectionOptions::new()
-        .with_name("nats-box rust example")
-        .connect(&args.server)?;
+    let nc: nats::Connection;
+
+    if args.creds.len() > 0 {
```

Comment:
can creds be an `Option<String>` to avoid conflating emptiness with presence?

Reply:
Not sure, depends on CLI framework.

Regardless of that, what would your optimum connection constructor look like here with optional creds?

Reply:
I'd recommend doing this: https://github.com/nats-io/nats.rs/blob/3c8e91690d337f3cbe499e5a48a46288216cae0d/examples/nats_bench.rs#L19-L21

Reply:
That is why I asked, I like it! I will make the changes.. Thanks!

Reply:
Please take another look, thanks again for the suggestion, exactly what I was looking for.